gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -106,13 +105,10 @@ def self._check_final_ancestors(target, target_ancestors, source_method_names)
     # use reverse_each to check farther-up ancestors first, for better error messages. we could avoid this if we were on
     # the version of ruby that adds the optional argument to method_defined? that allows you to exclude ancestors.
     target_ancestors.reverse_each do |ancestor|
-      next if !module_with_final?(ancestor)
+      final_methods = @modules_with_final.fetch(ancestor, nil)
+      next if !final_methods
```

Comment:
This includes the exception that was outlined above where a `nil` entry means that the entry has final methods somewhere along its ancestor chain. Is it OK to continue in that case? If so, could you include a comment that explains why?

Reply:
It's OK to continue because either the ancestor has `final` methods somewhere along its ancestor chain, but no methods in itself, or the ancestor doesn't have `final` methods along its ancestor chain.  Either way, there are no methods to check for `final` violations in ancestor itself.  I'll add a comment.